platforms/documentation/docs/src/docs/userguide/best-practices/bp_tasks.adoc

```diff
@@ -68,3 +69,55 @@ Gradle can use this information to optimize task scheduling.
 
 === Tags
 `#tasks`, `#inputs-and-outputs`, `#up-to-date-checking`
+
+[[use_cacheability_annotations]]
+== Favor Cacheability Annotations Over Cache Control Methods
+
+The link:{javadocPath}/org/gradle/api/tasks/TaskOutputs.html#cacheIf(org.gradle.api.specs.Spec)[`cacheIf`] and link:{javadocPath}/org/gradle/api/tasks/TaskOutputs.html#doNotCacheIf(java.lang.String,org.gradle.api.specs.Spec)[`doNotCacheIf`] methods should only be used in situations when the cacheability of a task can vary between different task instances, or can not be determined until the task is configured by Gradle.
+
+You should instead favor annotating the task class itself with the link:{javadocPath}/org/gradle/api/tasks/CacheableTask.html[`@CacheableTask`] annotation for any task that is meant to _always_ be cacheable.
+Likewise, the link:{javadocPath}/org/gradle/work/DisableCachingByDefault.html[`@DisableCachingByDefault`] should be used to always disable caching for all instances of a task type.
+
+Remember that only tasks that produce reproducible and relocatable output should be marked as `@Cacheable`.
+
+=== Explanation
+
+Annotating a task type will ensure that _each task instance_ of that type is properly understood by Gradle to be cacheable (or not cacheable).
+This removes the need to remember to configure each of the task instances separately in buildscripts.
+Using the annotations also _documents_ the intended cacheability of the task type within its own source, appearing in Javadoc and making the task's behavior clear to other developers without requiring them to inspect each task instance's configuration.
+It is also slightly more efficient than running a test at configuration time to determine cacheability.
+
+=== Example
+
+==== Don't Do This
+
+If we want to reuse the output of a task, we shouldn't do this:
```

Comment:
styling question, should we write in plural **first-person** form (i.e. ..we want..) ? or is it acceptable to write in second-person form (i.e. ...you want...)?

Reply:
I'm going to merge this now, but please do answer.  I'll update the language in this and the other item as needed.

Reply:
Both **first-person** plural ("we") and **second-person** ("you") are acceptable. 

The best choice depends on the tone and style of your documentation. It's also about consistency (whatever we do here we must do in all best practices docs).

I personally prefer "you" for this type of documentation since it feels more like a How-To-Guide. **You** is more direct and instructional.

Reply:
Sounds good to me.  I'll update this to use "you".